File: securedoors.py
def entry_list():
    log =[]
    n = input()
    for e in range(int(n)):
        line = input().split()
        action = line[0]
        name = line[1]

        if action == "entry":
            if name not in log:
                print(name + " entered")
                log.append(name)
            else:
                print(name + " entered (ANOMALY)") 
        else:
            if name not in log:
                print(name + " exited (ANOMALY)")
            else:
                print(name + " exited")
                log.remove(name)

entry_list()

File: securedoors.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<string> names (n);

    for (int i=0; i < n; i++) {
        string action, name;
        cin >> action >> name;
        if (action == "entry" && names[name]) {
        }
    }
}